staging/kos/pkg/controllers/ipam/ipam.go

```diff
@@ -780,27 +782,47 @@ func (ctlr *IPAMController) pickAndLockAddress(ns, name string, att *netv1a1.Net
 }
 
 func (ctlr *IPAMController) updateNAStatus(ns, name string, att *netv1a1.NetworkAttachment, nadat *NetworkAttachmentData, statusErrs []string, subnetUID k8stypes.UID, lockForStatus ParsedLock, ipForStatus gonet.IP) error {
-	att2 := att.DeepCopy()
-	att2.Status.Errors.IPAM = statusErrs
-	att2.Status.LockUID = string(lockForStatus.UID)
-	att2.Status.AddressVNI = lockForStatus.VNI
+	test, _ := ctlr.netattLister.NetworkAttachments(ns).Get(name)
+	if test == nil { // It has been deleted, don't bother
+		return nil
+	}
+	dStatus := netv1a1.NetworkAttachmentStatus{
+		Errors:     netv1a1.NetworkAttachmentErrors{IPAM: statusErrs},
+		LockUID:    string(lockForStatus.UID),
+		AddressVNI: lockForStatus.VNI,
+	}
+	if ipForStatus != nil {
+		dStatus.IPv4 = ipForStatus.String()
+	}
+	var pb bytes.Buffer
+	encoder := json.NewEncoder(&pb)
+	err := encoder.Encode(dStatus)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to Encode(%#+v): %#+v", dStatus, err))
+	}
+	patchString := pb.String()
+	if patchString[0] != '{' {
+		panic(patchString)
+	}
+	if len(statusErrs) == 0 {
```

Comment:
Isn't a check like this needed for `lockForStatus.UID` as well?

Reply:
Yeah, and the VNI too.